src/hooks/useDisableInputUntilReply.ts

```diff
@@ -38,12 +47,7 @@ export const useDisableInputUntilReply = ({
         setIsMessageInputDisabled(false);
       }
     }
-    return () => {
-      if (timerRef.current) {
-        clearTimeout(timerRef.current);
-      }
-    };
-  }, [currentUserId, botUser, lastMessage]);
+  }, [currentUserId, botUser, lastMessage, blockInputWhileBotResponding]);
```

Comment:
```suggestion
  }, [currentUserId, botUser?.userId, lastMessage?.messageId, blockInputWhileBotResponding]);
```

Better to keep the items in string format since React uses shallow equality comparison to identify whether dependencies are updated.

Reply:
in this case, `lastMessage` is intended becaue of streaming. messageId does not change nor updatedTs. It is message and streaming state